server/src/main/java/org/apache/iotdb/db/newsync/sender/pipe/TsFilePipeData.java

```diff
@@ -0,0 +1,198 @@
+    this.tsFilePath = tsFilePath;
+    this.serialNumber = serialNumber;
+  }
+
+  public TsFilePipeData(Deletion deletion, long serialNumber) {
+    this.deletion = deletion;
+    this.serialNumber = serialNumber;
+  }
+
+  public TsFilePipeData(PhysicalPlan plan, long serialNumber) {
+    this.plan = plan;
+    this.serialNumber = serialNumber;
+  }
+
+  public boolean isTsFile() {
+    return tsFilePath != null;
+  }
+
+  public boolean isTsFileClosed() {
+    File tsFile = new File(tsFilePath).getAbsoluteFile();
+    File resource = new File(tsFile.getAbsolutePath() + TsFileResource.RESOURCE_SUFFIX);
+    return resource.exists();
+  }
+
+  public String getTsFilePath() {
+    return tsFilePath == null ? "null" : tsFilePath;
+  }
+
+  public long getSerialNumber() {
+    return serialNumber;
+  }
+
+  //    public Loader.Type getLoaderType() {
+  //      if (tsFilePath != null) {
+  //        return Loader.Type.TsFile;
+  //      } else if (deletion != null) {
+  //        return Loader.Type.Deletion;
+  //      } else if (plan != null) {
+  //        return Loader.Type.PhysicalPlan;
+  //      }
+  //      logger.error("Wrong type for transport type.");
+  //      return null;
+  //    }
+
+  public List<File> getTsFiles() throws FileNotFoundException {
+    File tsFile = new File(tsFilePath).getAbsoluteFile();
+    File resource = new File(tsFile.getAbsolutePath() + TsFileResource.RESOURCE_SUFFIX);
+    File mods = new File(tsFile.getAbsolutePath() + ModificationFile.FILE_SUFFIX);
+
+    List<File> files = new ArrayList<>();
+    if (!tsFile.exists()) {
+      throw new FileNotFoundException(String.format("Can not find %s.", tsFile.getAbsolutePath()));
+    }
+    files.add(tsFile);
+    if (resource.exists()) {
+      files.add(resource);
+    }
+    if (mods.exists()) {
+      files.add(mods);
+    }
+    return files;
+  }
+
+  public byte[] getBytes() {
+    ByteBuffer buffer = ByteBuffer.allocate(SERIALIZE_BUFFER_SIZE);
+    if (deletion != null) {
+      deletion.serializeWithoutFileOffset(buffer);
+    } else if (plan != null) {
+      plan.serialize(buffer);
+    } else if (tsFilePath != null) {
+      ReadWriteIOUtils.write(tsFilePath, buffer);
+    } else {
+      logger.error(String.format("Null tsfile pipe data with serialNumber %d.", serialNumber));
+    }
+
+    byte[] bytes = new byte[buffer.position()];
+    buffer.flip();
+    buffer.get(bytes);
+    return bytes;
+  }
+
+  public long serialize(DataOutputStream stream) throws IOException {
+    long serializeSize = 0;
+
+    stream.writeLong(serialNumber);
+    serializeSize += Long.BYTES;
+    if (tsFilePath != null) {
+      stream.writeByte((byte) Type.TSFILE.ordinal());
+    } else if (deletion != null) {
+      stream.writeByte((byte) Type.DELETION.ordinal());
+    } else if (plan != null) {
+      stream.writeByte((byte) Type.PHYSICALPLAN.ordinal());
+    } else {
+      logger.error(String.format("Null tsfile pipe data with serialNumber %d.", serialNumber));
```

Comment:
`serializeSize += Byte.BYTES` but actually it doesn't write anything if  null tsfile pipe data

Reply:
fixed